### PR TITLE
Introduce new test scripts

### PR DESCRIPTION
**Pull request by LPGhatguy**
This PR enables us to run TestEZ's tests in Studio, with Lemur, and with Roblox-CLI (currently internal only). It's based off of a [Roact-Rodux PR](https://github.com/Roblox/roact-rodux/pull/25) I filed a few days ago.

This PR also upgrades TestEZ's miniature self-test testing framework to have more output and integrate it with our other running environments.

### CODE REVIEW

File: test/runner.server.lua [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		local tests = require(testModule)
+
+		print(string.format("%s", testModule.Name))
+
+		for testName, testFunction in pairs(tests) do
+			local success, message = pcall(testFunction)
+
+			if success then
+				print(string.format("  [PASS] %s", testName))
+				successCount = successCount + 1
```
Comment (ZoteTheMighty) [inline test/runner.server.lua]: Are you using this count anywhere?

File: test/runner.server.lua [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			else
+				print(string.format("  [FAIL] %s", testName))
+				failureCount = failureCount + 1
```
Comment (ZoteTheMighty) [inline test/runner.server.lua]: Same as above; looks like all we need to know here is that something failed at all.

Comment (ZoteTheMighty) [reply test/runner.server.lua]: Looks like most of this was adapted from the previous `spec.lua` file, but we dropped the print at the end with the number of successes and failures.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -19,21 +19,31 @@ To get started working on TestEZ, you'll need:
...
+lua -lluacov test/lemur.lua
 luacov
 ```
 
+If you're an engineer at Roblox, you can skip this setup and use Roblox-CLI. Make sure it's on your `PATH` and that you have a production Roblox Studio installation. You can then run:
+
+```sh
+./test/roblox-cli.sh
+```
+
```
Comment (benbrimeyer) [inline CONTRIBUTING.md]: Should we add instructions on how to run these tests in Studio since that target is also supported in this change?

Comment (LPGhatguy) [reply CONTRIBUTING.md]: That's a good idea! This guide is duplicated in a lot of repositories, I wish we had a better system for propagating fixes like that.

**Comment by LPGhatguy** [comment]
Circling back to this PR, just merged in the changes that came out of the TestEZ CLI, including switching to GitHub Actions.

**Pull request merged**